### ISSUE TITLE
"powered by php"

### ISSUE THREAD

**Issue by bates64** [SharpScratchMod/sharp-www#1]
:-1:

[take](https://github.com/OpenSprites/OpenSprites-Origin) [opensprites](https://github.com/OpenSprites/OpenSprites-next) [for](https://github.com/OpenSprites/OpenSprites-django) [example](https://eev.ee/blog/2012/04/09/php-a-fractal-of-bad-design/)

**Comment by lholliger** [comment]
PHP is the best for me :)

**Comment by Mrcomputer1** [comment]
PHP is just easy for creating stuff in

We might try changing to Django (Python) or Node after we have a working site (in PHP)

**Comment by bates64** [comment]
> PHP is just easy for creating stuff in

And the moment the site gets larger, it becomes horribly slow and hard to maintain :tada:

**Comment by ghost** [comment]
I know...

**Comment by ghost** [comment]
We might move it from PHP later

**Comment by ghost** [comment]
But for now it is the best option we have

**Comment by bates64** [comment]
> it is the best option we have

how

**Comment by ghost** [comment]
ok,  i don't know how to use Node.JS as a backend, and with Python it is the same

**Comment by bates64** [comment]
learn node then

it isn't hard:

``` js
var http = require('http');

http.createServer(function(req, res) {
  res.end('topkek');
}).listen(1234);
```

**Comment by Mrcomputer1** [comment]
The HTTP module is hard when you add POST requests, query strings, cookies, databases, uploads and things like that.

Things like http://expressjs.com/ and http://feathersjs.com/ don't work well for me.
NPM just does a timeout error

---

Also I have a kind of odd plan:
We are going to make sharp-www in PHP like ScratchR (Scratch's 1.x site) was made in PHP
Then we are going to rewrite sharp-www in Python with Django like ScratchR2 (Scratch 2.x site) was made in Python with Django
Then we are going to rewrite sharp-www in Node and maybe Express and React like scratch-www (New scratch site) was made in Node, express and react

**Comment by ghost** [comment]
@nanalan i have

**Comment by bates64** [comment]
> Things like http://expressjs.com/ and http://feathersjs.com/ don't work well for me.
> NPM just does a timeout error

...that's something totally wrong with your system then.

``` sh
$ npm install --save express
```

works totally fine for the best of us.

> Also I have a kind of odd plan:
> We are going to make sharp-www in PHP like ScratchR (Scratch's 1.x site) was made in PHP
> Then we are going to rewrite sharp-www in Python with Django like ScratchR2 (Scratch 2.x site) was made in Python with Django
> Then we are going to rewrite sharp-www in Node and maybe Express and React like scratch-www (New scratch site) was made in Node, express and react

...
you're basically _planning_ to rewrite
wat

**Comment by Mrcomputer1** [comment]
> ...that's something totally wrong with your system then.

More like wrong with my internet. Some sites it will timeout on and others not. The NPM registry happens to be one of those sites. Only some of the NPM packages will trigger a timeout error.

> you're basically _planning_ to rewrite

Only after we get bored of using/making the PHP site :P

**Comment by ghost** [comment]
http://www.djangobook.com/en/2.0/introduction.html:

> PHP fixed many of these problems, and it took the world by storm – it’s now by far the most popular tool used to create dynamic Web sites, and dozens of similar languages and environments (ASP, JSP, etc.) followed PHP’s design closely. PHP’s major innovation is its ease of use: PHP code is simply embedded into plain HTML; the learning curve for someone who already knows HTML is extremely shallow.
> 
> But PHP has its own problems; its very ease of use encourages sloppy, repetitive, ill-conceived code. Worse, PHP does little to protect programmers from security vulnerabilities, and thus many PHP developers found themselves learning about security only once it was too late.
> gotta agree with that 
> we'll stick with PHP for now, but soon (not now! :stuck_out_tongue: ) we might change to django
